### PR TITLE
Missing element in page becomes an exception

### PR DESCRIPTION
**Pull request by giorgiosironi**
I have seen this `Notice` while using the SDK in search, so this makes it into a more manageable error and includes some debugging information.

Exception class is just a standard one, but open to suggestions

### CODE REVIEW

File: src/SlicedIterator.php [hunk 1]
```diff
@@ -13,7 +15,11 @@
         $page = ceil($this->key / $this->pageBatch);
         $inPage = $this->key - ($page * $this->pageBatch) + $this->pageBatch - 1;
 
-        return $this->getPage($page)[$inPage];
+        $pageContents = $this->getPage($page);
+        if (!array_key_exists($inPage, $pageContents)) {
+            throw new LogicException("Cannot find element with key $inPage in page $page");
```
Comment (thewilkybarkid) [inline src/SlicedIterator.php]: `OutOfRangeException`?

**Comment by thewilkybarkid** [comment]
Are the problems caused by the data? (It shouldn't be possible to hit this...)

**Comment by giorgiosironi** [comment]
Don't know what is causing the problem but all tests are failing after the change... Will debug this.

**Comment by thewilkybarkid** [comment]
`ceil()` returns a `float`. Just use `isset()` instead.

**Comment by giorgiosironi** [comment]
I believe `isset()` is too wide as `isset($pageContentAsANonExistentVariable[1])` while `array_key_exists()` should give the error `Undefined variable`

**Comment by thewilkybarkid** [comment]
`$this->getPage($page)` _has_ to return an array.

**Comment by thewilkybarkid** [comment]
There is a reason not to use `isset()` at the minute: we have `null` values when skipping invalid arguments.

**Pull request merged**